### PR TITLE
Activities

### PR DESCRIPTION
**Pull request by phamkykhoi**

### CODE REVIEW

File: app/controllers/relationships_controller.rb [hunk 1]
```diff
@@ -6,7 +6,7 @@ def create
   end
 
   def destroy
-    @user = Relationship.(find params[:id]).followed
+    @user = Relationship.find(params[:id]).followed
     current_user.unfollow(@user)
```
Comment (nguyenduyta) [inline app/controllers/relationships_controller.rb]: remove ()

Comment (dieunb) [inline app/controllers/relationships_controller.rb]: remove `()`

File: app/models/concerns/activity_logs.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+module ActivityLogs
+  def create_activity user_id, target_id, status
+  	Activity.create! user_id: user_id, target_id: target_id, status: status
```
Comment (dieunb) [inline app/models/concerns/activity_logs.rb]: indent

File: app/models/lesson.rb [hunk 3]
```diff
@@ -1,18 +1,22 @@
...
+
   delegate :fullname, to: :user
   delegate :name, to: :category, prefix: true
 
   scope :lesson_user, ->user {where("lessons.user_id IN
     (SELECT followed_id FROM relationships WHERE follower_id = '#{user.id}')
     OR lessons.user_id = '#{user.id}'")}
-  
+
+  scope :lesson_learned, ->user {(where user_id: user.id)}
```
Comment (dieunb) [inline app/models/lesson.rb]: remove `()`

File: app/models/relationship.rb [hunk 4]
```diff
@@ -1,7 +1,22 @@
...
   validates :follower_id, presence: true
   validates :followed_id, presence: true
+
+  after_save :follow_activity
+  before_destroy :unfollow_activity
+
+  private
+
+  def follow_activity
+  	create_activity follower_id, followed_id , "follow"
```
Comment (dieunb) [inline app/models/relationship.rb]: indent

File: app/models/relationship.rb [hunk 5]
```diff
@@ -1,7 +1,22 @@
...
+  end
+
+  def unfollow_activity
+  	create_activity follower_id, followed_id , "unfollow"
```
Comment (dieunb) [inline app/models/relationship.rb]: indent

File: app/models/concerns/activity_logs.rb (hunk 2 above)
Comment (nguyenduyta) [inline app/models/concerns/activity_logs.rb]: indent

File: app/models/lesson.rb (hunk 3 above)
Comment (nguyenduyta) [inline app/models/lesson.rb]: remove

File: app/models/relationship.rb [hunk 6]
```diff
@@ -1,7 +1,22 @@
 class Relationship < ActiveRecord::Base
+	include ActivityLogs
```
Comment (nguyenduyta) [inline app/models/relationship.rb]: indent

File: app/models/relationship.rb (hunk 4 above)
Comment (nguyenduyta) [inline app/models/relationship.rb]: indent

**Pull request merged**